eventmesh/ApiServerEventMesh.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const assert = require('assert');
+const config = require('../common/config');
+const logger = require('../common/logger');
+const CONST = require('../common/constants');
+const EventMeshServer = require('./EventMeshServer');
+const kc = require('kubernetes-client');
+const JSONStream = require('json-stream');
+
+
+const apiserver = new kc.Client({
+  config: {
+    url: `https://${config.internal.ip}:9443`,
+    insecureSkipTlsVerify: true
+  },
+  version: '1.9'
+});
+
+class ApiServerEventMesh extends EventMeshServer {
+  registerWatcher(resourceName, resourceType, callback) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => {
+        const stream = apiserver
+          .apis[`${resourceName}.servicefabrik.io`]
+          .v1alpha1.watch.namespaces('default')[resourceType].getStream();
+        const jsonStream = new JSONStream();
+        stream.pipe(jsonStream);
+        jsonStream.on('data', callback);
+      })
+      .catch(e => logger.error('Caucht error while registering', e));
+  }
+  createLockResource(name, type, body) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver.apis[`${name}.servicefabrik.io`].v1alpha1.namespaces('default')[type].post({
```

Comment:
constants

`'default'` namespace
`.servicefabrik.io` host

Reply:
Catch with error code and throw errors defined in `errors` class